lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -40,14 +40,28 @@ def get(name, rgroup = configuration.resource_group)
         model_class.new(response)
       end
 
+      # Delete the resource with the given +name+ for the provided +resource_group+,
+      # or the resource group specified in your original configuration object. If
+      # successful, returns a ResponseHeaders object.
+      #
+      # If the delete operation returns a 204 (no body), which is what the Azure
+      # REST API typically returns if the resource is not found, it is treated
+      # as an error and a ResourceNotFoundException is raised.
+      #
       def delete(name, rgroup = configuration.resource_group)
         validate_resource_group(rgroup)
         validate_resource(name)
 
         url = build_url(rgroup, name)
         url = yield(url) || url if block_given?
-        rest_delete(url)
-        nil
+        response = rest_delete(url)
+
+        if response.code == 204
+          msg = "#{self.class} resource #{rgroup}/#{name} not found"
+          raise Azure::Armrest::ResourceNotFoundException.new(response.code, msg, response)
+        end
+
+        Azure::Armrest::ResponseHeaders.new(response.headers)
```

Comment:
I read the original PR #203 and understood the use of response header to pull status of the asynchronous call. 
If the only use is for asynchronous call, can we create a new class to wrap the header more properly, and provide one only method to report the status of the asynchronous call?